### DISCUSSION TITLE
Cache invalidation without an update

### DISCUSSION THREAD

**Discussion by Psvensso** [urql-graphql/urql#3049] [Q&A]
Were trying to evict some entities in our persisted graphcache to save storage on the users device.
Lets imagine a "Download playlist for offline use" scenario. Were trying to handle the "De-select offline playlist", 
this is in my mind a clientside only scenario where we just want to evict the playlist and it's songs or videos from the cache but there has not been a mutation and the state on the server has not changed. 

 The closest iv'e been to success is to call a "evict" mutation and handle the cache in the update of that mutation but it feels wrong, the mutation does nothing and i can't get the mutation to accept the "cache-only" policy. 
 
 Any other takes on how to do this?

**Comment by JoviDeCroock** [comment]
I wonder why `Download playlist for offline use` does need a mutation while the opting out doesn't. I guess it isn't relevant to i.e. the subscription status of the user to track whether they have downloaded stuff present? The main thinking behind always needing a mutation in graphCache is that it aims to reflect server-side state so it makes me wonder whether the downloaded songs etc should be part of graphCache in that case

**Reply by Psvensso** [reply]
Well this was just an perhaps bad example. 
What we need anyway is to locally evict data from the client, the state has not changed hence there is no mutation to call we just dont want to keep that state locally anymore. 
I understand the use-case for always keeping the data in sync and reflecting the state of the server, in this case we dont want to alter the records, we just want to clear out cached queries. Without resetting the entire state.

**Reply by JoviDeCroock** [reply]
The react-native storage addon uses a TTL, type of thing this might be interesting in this context https://github.com/urql-graphql/urql/blob/main/packages/storage-rn/src/makeAsyncStorage.ts

A similar concept could be used to selectively filter out stuff while reading/writing
